app/controllers/auth_services_controller.rb

```diff
@@ -70,4 +73,25 @@ def omniauth_providers
   def redirect_path
     :services
   end
+
+  def is_viable_redirect_location(url)
+    # When you visit an attendable page (e.g. a Workshop, Eventor  Meeting) you
+    # probably want to sign-up to that thing, which requires signing in. When
+    # you sign in you want to get sent back to the page you were on previously
+    # so you can click the attend button right away rather than being sent to
+    # your dashboard then having to navigate back to the Workshop/Event/Meeting
+    # page.
+    begin
+      route = Planner::Application.routes.recognize_path(url)
```

Comment:
@nickcharlton Hey, you mentioned this got deprecated. Do you have another solution as I really want to get this merged in?